WalletWasabi.Fluent/Behaviors/AutoPasteAmountBehavior.cs

```diff
@@ -0,0 +1,84 @@
+using System.Globalization;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using WalletWasabi.Fluent.Controls;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Fluent.Behaviors;
+
+public class AutoPasteAmountBehavior : AttachedToVisualTreeBehavior<DualCurrencyEntryBox>
+{
+	protected override void OnAttachedToVisualTree(CompositeDisposable disposable)
+	{
+		if (AssociatedObject is null)
+		{
+			return;
+		}
+
+		if (!Services.UiConfig.AutoPaste)
+		{
+			return;
+		}
+
+		AutoPaster(AssociatedObject, "PART_LeftEntryBox", IsValidBtc)
+			.DisposeWith(disposable);
+
+		AutoPaster(AssociatedObject, "PART_RightEntryBox", IsValidUsd)
+			.DisposeWith(disposable);
+	}
+
+	private static bool IsValidUsd(string s)
+	{
+		var canParse = decimal.TryParse(s, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var d);
+		return canParse && CountDecimalPlaces(d) <= 2 && d > 0;
+	}
+
+	private static bool IsValidBtc(string s)
+	{
+		var canParseAsDecimal = decimal.TryParse(s, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var d);
+
+		if (canParseAsDecimal)
+		{
+			var lessThanMaximum = d < Constants.MaximumNumberOfSatoshis;
```

Comment:
Here you should compare with the maximum number in bitcoins and not satoshis, because currently we enter only amounts in btc.

```suggestion
			var lessThanMaximum = d < Constants.MaximumNumberOfBitcoins;
```